### PR TITLE
sort by date without obeying folder/files

### PR DESCRIPTION
**Pull request by tobiasKaminsky**
ref: #1334

### CODE REVIEW

**Comment by tobiasKaminsky** [comment]
sort

**Comment by tobiasKaminsky** [comment]
@AndyScherzinger @davivel @przybylski please review

File: src/com/owncloud/android/utils/FileStorageUtils.java [hunk 1]
```diff
@@ -184,15 +184,7 @@ public static RemoteFile fillRemoteFile(OCFile ocFile){
...
-                if (o1.isFolder() && o2.isFolder()) {
-                    Long obj1 = o1.getModificationTimestamp();
-                    return val * obj1.compareTo(o2.getModificationTimestamp());
-                }
-                else if (o1.isFolder()) {
-                    return -1;
-                } else if (o2.isFolder()) {
-                    return 1;
-                } else if (o1.getModificationTimestamp() == 0 || o2.getModificationTimestamp() == 0){
+                if (o1.getModificationTimestamp() == 0 || o2.getModificationTimestamp() == 0){
```
Comment (przybylski) [inline src/com/owncloud/android/utils/FileStorageUtils.java]: I don't really like that if there is no timestamp we artificially making it equal with everything. This leads to unexpected behavior when we do have multiple zeros in collection.
I would rather not take any special care of this and boil down compare method to something like this:
```java
public int compare(OCFile o1, OCFile o2) {
    return val * (o1.getModificationTimestamp() - o2.getModificationTimestamp());
}
```

Comment (tobiasKaminsky) [inline src/com/owncloud/android/utils/FileStorageUtils.java]: Yeah, but this is already the current behaviour.
So, we should discuss this in another issue?
(Although I do not know how often this problem might occur as the modification timestamp should always be set, or?

Comment (przybylski) [inline src/com/owncloud/android/utils/FileStorageUtils.java]: In that case its ok for me to be merged

**Comment by abelgardep** [comment]
To be done in #2710

**Pull request closed without merging**